ui/app/src/state/reducers/dialogStack.ts

```diff
@@ -27,6 +28,39 @@ const reducer = createReducer<GlobalState['dialogStack']>(
 	},
 	(builder) => {
 		builder.addCase(pushDialog, (state, { payload }) => {
+			const component = payload.component;
+			// If the dialog is a CodeEditorDialog, FormsEngineDialog or LegacyFormDialog, we run extra verification.
+			// If the same type of dialog is already open for the same path, we update its props instead of creating a new
+			// dialog, and reset the isMinimized state.
+			if (
+				component === 'craftercms.components.CodeEditorDialog' ||
+				component === 'craftercms.components.LegacyFormDialog' ||
+				component === 'craftercms.components.FormsEngineDialog'
+			) {
+				const dialogState = Object.values(state.byId).find((dialog) => {
+					if (dialog.component === 'craftercms.components.FormsEngineDialog') {
+						return (
+							dialog.component === component &&
+							(dialog.props as FormsEngineDialogProps).formProps.update?.path ===
+								(payload.props as FormsEngineDialogProps).formProps.update?.path
+						);
+					} else {
+						// @ts-expect-error TS2339: Props may be type CodeEditorDialogProps or LegacyFormDialogProps.
+						return dialog.component === component && dialog.props.path === payload.props.path;
```

Comment:
Use the types accordingly, then...
```suggestion
            const props: CodeEditorDialogProps | LegacyFormDialogProps = payload.props;
						return dialog.component === component && dialog.props.path === props.path;
```

Always aim for strong typing.